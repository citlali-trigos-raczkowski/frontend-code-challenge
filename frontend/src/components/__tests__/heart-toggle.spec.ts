import { shallowMount, createLocalVue } from "@vue/test-utils";
import HeartToggle from "../heart-toggle.vue";
import VueRouter from "vue-router";
import { HeartToggleMockData } from "../../../tests/mocks/mock-props";

const localVue = createLocalVue();
localVue.use(VueRouter);

describe("component/heart-toggle.vue", () => {
  it("renders correctly", () => {
    const wrapper = shallowMount(HeartToggle, {
      propsData: HeartToggleMockData,
      localVue,
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it("color sets as expected", () => {
    const wrapper = shallowMount(HeartToggle, {
      propsData: HeartToggleMockData,
      localVue,
    });

    const aWrapper = wrapper.find("v-btn-stub");
    expect(aWrapper.attributes().color).toBe(HeartToggleMockData.heartColor);
  });
});
