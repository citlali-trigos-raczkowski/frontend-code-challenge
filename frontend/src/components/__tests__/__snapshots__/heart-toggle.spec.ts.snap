// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component/heart-toggle.vue renders correctly 1`] = `
<v-card-title-stub
  id="tile-header"
>
  <v-col-stub
    id="tile-heart-container"
    tag="div"
  >
    <v-btn-stub
      activeclass=""
      color="pink"
      data-cy="toggle-pokemon-favorite"
      icon="true"
      id="tile-heart"
      tag="button"
      type="button"
    >
      <v-icon-stub>
        mdi-heart
      </v-icon-stub>
    </v-btn-stub>
  </v-col-stub>
</v-card-title-stub>
`;
